pkg/auth/kubeconfig.go

```diff
@@ -185,37 +211,127 @@ func generateX509KubeConfig(ctx context.Context, cli kubernetes.Interface, cfg *
 
 	csrBytes, err := x509.CreateCertificateRequest(rand.Reader, template, privateKey)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	csrPemBytes := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE REQUEST", Bytes: csrBytes})
 	_, _ = fmt.Fprintf(writer, "Certificate request generated.\n")
+	return csrPemBytes, keyBytes, nil
+}
+
+func makeCSRName(user string) string {
+	return fmt.Sprintf("%s-%s-%s", CSRNamePrefix, user, uuid.NewString()[:8])
+}
 
+func generateX509KubeConfig(ctx context.Context, cli kubernetes.Interface, cfg *clientcmdapi.Config, writer io.Writer, opts *KubeConfigGenerateX509Options) (*clientcmdapi.Config, error) {
```

Comment:
Is it possible to use the functions inside 
https://github.com/kubernetes/client-go/blob/master/util/certificate/csr/csr.go ? I think it handles the reuse of existing CSR & the compatibility problem for CSR v1&v1beta1. Would you mind having a try for that?

Reply:
It is okay to leave it for future enhancement. I can try to rework this part of logic later.